src/c++/perf_analyzer/test_request_rate_manager.cc

```diff
@@ -39,251 +39,295 @@ namespace triton { namespace perfanalyzer {
+      // noise. Allow 5% slop
+      //
+      CHECK(
+          delay_average ==
+          doctest::Approx(expected_delay_average).epsilon(0.05));
+      CHECK(delay_variance == doctest::Approx(delay_average).epsilon(0.05));
+    } else if (request_distribution == CONSTANT) {
+      // constant should in theory have 0 variance, but with thread timing
+      // there is obviously some noise.
+      //
+      // Allow it to be at most 5% of average
+      //
+      auto max_allowed_delay_variance = 0.05 * delay_average;
+
+      // Constant should be pretty tight. Allowing 1% slop there is noise in the
+      // thread scheduling
+      //
+      CHECK(
+          delay_average ==
+          doctest::Approx(expected_delay_average).epsilon(0.01));
+      CHECK_LT(delay_variance, max_allowed_delay_variance);
+    } else {
+      throw std::invalid_argument("Unexpected distribution type");
     }
+  }
 
-    std::vector<int64_t> GatherTimeBetweenRequests(const std::vector<std::chrono::time_point<std::chrono::system_clock>>& timestamps) {
-      std::vector<int64_t> time_between_requests;
+  void CheckSequences(PerfAnalyzerParameters params)
+  {
+    auto stats = GetStats();
 
-      for (size_t i = 1; i < timestamps.size(); i++) {
-        auto diff = timestamps[i] - timestamps[i-1];
-        std::chrono::nanoseconds diff_ns = std::chrono::duration_cast<std::chrono::nanoseconds>(diff);
-        time_between_requests.push_back(diff_ns.count());
-      }
-      return time_between_requests;
+    // Make sure all seq IDs are within range
+    //
+    for (auto seq_id : stats->sequence_status.used_seq_ids) {
+      CHECK(seq_id >= params.start_sequence_id);
+      CHECK(seq_id <= params.start_sequence_id + params.sequence_id_range);
     }
 
-    double CalculateAverage(std::vector<int64_t> values) {
-      double avg = std::accumulate(values.begin(), values.end(), 0.0) / values.size();
-      return avg;
+    // Make sure that we had the correct number of concurrently live sequences
+    //
+    // If the sequence length is only 1 then there is nothing to check because
+    // there are never any overlapping requests -- they always immediately exit
+    //
+    if (params.sequence_length != 1) {
+      CHECK(
+          params.num_of_sequences == stats->sequence_status.max_live_seq_count);
     }
 
-    double CalculateVariance(std::vector<int64_t> values, double average) {
-      double tmp = 0;
-      for (auto value : values) {
-        tmp += (value - average) * (value - average) / values.size();
-      }
-      double variance = sqrt(tmp);
-      return variance;
+    // Make sure that the length of each sequence is as expected
+    // (The code explicitly has a 20% slop, so that is what we are checking)
+    //
+    for (auto len : stats->sequence_status.seq_lengths) {
+      CHECK(len == doctest::Approx(params.sequence_length).epsilon(0.20));
+    }
+  }
+
+  std::vector<int64_t> GatherTimeBetweenRequests(
```

Comment:
I just realized there's a bunch of stuff in these diffs because of the new usage of the code formatter